File: Minimum Waiting Time/MinimumWaitingTime.java
import java.util.*;

class MinimumWaitingTime {

    public int minimumWaitingTime(int[] queries) {
        if (queries.length == 1)
            return 0;
        int totalWaitTime = 0;
        int currentWaitTime = 0;
        Arrays.sort(queries);
        for (int i = 1; i < queries.length; i++) {
            currentWaitTime += queries[i - 1];
            totalWaitTime += currentWaitTime;
        }
        return totalWaitTime;
    }
}

File: Minimum Waiting Time/MinimumWaitingTime.kt
package com.algoexpert.program

fun minimumWaitingTime(queries: MutableList<Int>): Int {
    if (queries.size == 1) return 0
    var currentWaitingTime = 0
    var totalWaitingTime = 0
    queries.sort()
    for (i in 1 until queries.size) {
        currentWaitingTime += queries[i - 1]
        totalWaitingTime += currentWaitingTime
    }
    return totalWaitingTime
}
